packages/react/src/app-layout/AppLayoutSidebarNav.tsx

```diff
@@ -98,7 +112,16 @@ function AppLayoutSidebarNavListItem({
 				isActive={activePath === item.href}
 				hasEndElement={Boolean(endElement)}
 			>
-				<Link aria-current={active ? 'page' : undefined} {...restItemProps}>
+				<Link
+					aria-current={active ? 'page' : undefined}
+					{...restItemProps}
+					// Ignoring typescript here because `LinkProps` does not currently support `onClick`
+					// The code will run fine though, as link props are spread
+					// TODO we should add support for `onClick` in `LinkProps` and remove this `@ts-ignore`
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					onClick={onClick}
```

Comment:
@stowball we can talk further about this, but I have a follow up PR to this which will address this issue. The code runs fine, but typescript does not agree.